### ISSUE TITLE
libssh2_userauth_keyboard_interactive_ex callback and memory allocation

### ISSUE THREAD

**Issue by bagder** [libssh2/libssh2#1078]
**Describe the bug**

The function `libssh2_userauth_keyboard_interactive_ex` calls a function in the app to return a set of passwords.

These passwords are documented as: `String values will be free(3)ed by the library`.

This is a problem on (for example) Windows when libssh2 is used as a DLL, as then they use memory models that cannot be shared between an application and the DLL.

This is happens on git master and this issue has probably always existed. It can be circumvented by providing custom memory functions using `libssh2_session_init_ex`.

**Comment by vszakats** [comment]
It's surely confusing and non-obvious, but unless we want to change the API to never free() or return allocated pointers (to be freed by callers), we might make this gotcha clearer in the docs.

**Comment by bagder** [comment]
Yeah, I don't think we can fix this without adding a replacement function, and until someone wants to work on that I think clarifying this caveat in the documentation is the next best thing. At least there is a functioning workaround!
